### ISSUE TITLE
DELETE on multiple shards natively works

### ISSUE THREAD

**Issue by ozgune** [citusdata/citus#1130]
When users run a ``DELETE`` command that touches multiple shards, Citus currently displays an error message.  In #905, we introduced an improvement to our error messages to instruct the user to call ``master_modify_multiple_shards()`` for hash distributed tables, and ``master_apply_delete_command()`` for append distributed ones.

This issue tracks natively intercepting the ``DELETE`` command and having it call Citus' UDFs under the covers. We have two use cases for this scenario:

1. DELETE for expiring time-series data: Users who shard their data by tenant_id or entity_id (hash distribution) call DELETE to expire old events data. Users may also choose to call DELETE on append distributed tables, but this is likely out of scope.
2. DELETE a single record for a given tenant: Users can call this in one of two ways.
    1. DELETE ... WHERE tenant_id = 97 AND rec_id = 1; Citus already supports this.
    2. DELETE ... WHERE rec_id = 1; In this case, the user intends to do the above, but forgets to include tenant_id. As a result, they may run into small concurrency issues due to how we implement ``master_modify_multiple_shards()``. (Do they also accidentally delete data in this case?)

In summary, Citus could provide native DELETE support over multiple shards by calling ``master_modify_multiple_shards``. This may introduce small concurrency issues for users who make mistakes and forget to include the shard distribution column in their query.

**Comment by ozgune** [comment]
I'm noting #1131 as a semi-related issue. The underlying infrastructure for this issue and #1131 are about the same. However, the use cases and user's expectations are different.
